app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavouriteButton.kt

```diff
@@ -0,0 +1,57 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Color.Companion.Blue
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+data class FavouriteButtonState(
+    var isFavourite: Boolean
+)
+
+@Composable
+fun FavouriteButton(
+    buttonViewState: FavouriteButtonState,
+    modifier: Modifier=Modifier
+)
+{
+    var toggleIsFavourite by remember {
+        mutableStateOf(buttonViewState.isFavourite)
+    }
+    Box(
+        modifier = modifier
+            .padding(5.dp)
+            .size(37.dp)
+            .clip(shape = CircleShape)
+            .background(Color(0x0B253F).copy(alpha = 0.6F))
+            .clickable { toggleIsFavourite=!toggleIsFavourite },
+            contentAlignment = Alignment.Center
+    ){
+        Image(
+            painter = painterResource(id = if(toggleIsFavourite) R.drawable.favuoirteicon_selected else R.drawable.favouriteicon_unselected),
+            contentDescription = "Favourite Icon",
+            //modifier = modifier.clickable { toggleIsFavourite=!toggleIsFavourite }
```

Comment:
I commented it because upon trying to click the icon I found that sometimes it was a bit hard to click the small icon (inner QA coming through), so I moved the click logic to the whole Box element. Is this practice fine or should just the icon be clickable? Forgot to delete the commented code after moving the logic to the box.